views/js/QtiPreviewResultServerApi.js

```diff
@@ -44,9 +43,15 @@ define([
         var previewConsole = $('#preview-console');
         var variableIdentifier;
 
+        function escapeHtml(text) {
+            const div = document.createElement("div");
+            div.textContent = text;
+            return div.innerHTML;
+        }
```

Comment:
I realise you were tasked with removing lodash as much as possible, but I feel that adding local helpers like this is a code smell. (This one is a relatively simple function, so not the worst example.)

If we must have utility functions instead of lodash imports, shouldn't they all live in a common place?

I'm not even sure that the full removal is the smartest step, if it makes us end up with 10-20 "lodash-equivalent" helpers that we have to test and maintain.

I would prefer to see a PR that focuses first on removing just the incompatible usages between lodash 2 and 4 (I don't know how big that change would be, I'm guessing 15% of the size), and then we could go further with the vanilla Javascript replacements if desired.

Overall, this PR is too big to be properly reviewed. The changes could have been done in separate commits (first all the `_.each()`, then all the `_.assign()` etc.)